flask/app.py

```diff
@@ -790,25 +785,34 @@ def make_shell_context(self):
             rv.update(processor())
         return rv
 
-    def _reconfigure_for_run_debug(self, debug):
-        """The ``run`` commands will set the application's debug flag. Some
-        application configuration may already be calculated based on the
-        previous debug value. This method will recalculate affected values.
+    #: The environment value.  This is typically set from outside the
+    #: process by setting the `FLASK_ENV` environment variable and can be
+    #: used to quickly switch between different environments like
+    #: `production` and `development`.  If not set this defaults to
+    #: `production`.
+    env = ConfigAttribute('ENV')
 
-        Called by the :func:`flask.cli.run` command or :meth:`Flask.run`
-        method if the debug flag is set explicitly in the call.
+    def _get_debug(self):
+        return self.config['DEBUG']
+    def _set_debug(self, value):
+        self._set_debug_value(value)
 
-        :param debug: the new value of the debug flag
+    #: The debug flag.  Set this to ``True`` to enable debugging of the
+    #: application.  In debug mode the debugger will kick in when an unhandled
+    #: exception occurs and the integrated server will automatically reload
+    #: the application if changes in the code are detected.
+    #:
+    #: This attribute can also be configured from the config with the ``DEBUG``
+    #: configuration key.  Defaults to ``False``.
+    debug = property(_get_debug, _set_debug)
+    del _get_debug, _set_debug
 
-        .. versionadded:: 1.0
-            Reconfigures ``app.jinja_env.auto_reload``.
-        """
-        self.debug = debug
+    def _set_debug_value(self, value):
+        self.config['DEBUG'] = value
         self.jinja_env.auto_reload = self.templates_auto_reload
 
-    def run(
-        self, host=None, port=None, debug=None, load_dotenv=True, **options
-    ):
+    def run(self, host=None, port=None, debug=None,
+            load_dotenv=True, **options):
```

Comment:
I don't use this indentation style because it is visually inconsistent. This applies to a few places in this PR.

Reply:
What's wrong with it? It's pretty standard in Python ;)

Reply:
It creates inconsistent horizontal spacing. It gets especially weird when the method name is longer or the code is more indented.

Reply:
This has always been the pocoo codestyle.